src/com/amazon/ionschema/internal/util/RangeIonNumber.kt

```diff
@@ -35,7 +35,11 @@ internal class RangeIonNumber private constructor (
                 } else {
                     when (ion) {
                         is IonDecimal -> ion.bigDecimalValue()
-                        is IonFloat -> ion.bigDecimalValue()
+                        is IonFloat -> if (ion.isNumericValue) {
```

Comment:
For special values `+/- inf, nan`, `isNumericValue` is `false` and range doesn't include these special float values, they need to be added explicitly.

Reply:
Thank you (and Theo) for explaining. I wish we'd provided the method `isSpecialValue()` instead of `isNumericValue()`. It would be consistent with [phrasing from the spec](https://amzn.github.io/ion-docs/docs/float.html#special-values) and would avoid the overloaded phrase "numeric." Oh well.

Reply:
Could you add a clarifying comment to this branch? Something like:
```suggestion
                        is IonFloat -> if (ion.isNumericValue) {
                            // nan, +inf, -inf
```